### PR TITLE
Add missing files to makefiles, fixes make dist

### PR DESCRIPTION
**Pull request by drfiemost**
As the subject says, now I can create a tarball which compiles without errors.
Will probably need to add the contrib dir and other stuff too.

### CODE REVIEW

**Comment by kcgen** [comment]
Thanks for catching these discrepancies, @drfiemost.

What does your workflow look like involving `make dist` and creating a source tarball? 

The two major workflows involve building locally in ones git repo and using binaries generated by the CI system. It's always nice to know how others are building the package.

File: src/gui/Makefile.am [hunk 1]
```diff
@@ -6,5 +6,6 @@ libgui_a_SOURCES = sdlmain.cpp sdl_mapper.cpp dosbox_logo.h \
 	render_templates.h render_loops.h render_simple.h \
 	render_templates_sai.h render_templates_hq.h \
 	render_templates_hq2x.h render_templates_hq3x.h \
-	sdl_gui.cpp dosbox_splash.h render_glsl.h gui_msgs.h
+	sdl_gui.cpp render_glsl.h gui_msgs.h
 
+EXTRA_DIST = dosbox_staging_splash.c icon.c
```
Comment (kcgen) [inline src/gui/Makefile.am:11]: Add a newline.

File: src/dos/Makefile.am [hunk 2]
```diff
@@ -7,4 +7,4 @@ libdos_a_SOURCES = dos.cpp dos_devices.cpp dos_execute.cpp dos_files.cpp dos_ioc
 		   drives.cpp drive_virtual.cpp drive_local.cpp drive_cache.cpp drive_fat.cpp \
 		   drive_iso.cpp dev_con.h dos_mscdex.cpp dos_keyboard_layout.cpp \
 		   cdrom.h cdrom.cpp cdrom_image.cpp \
-		   drive_overlay.cpp
+		   drive_overlay.cpp program_autotype.h dos_mscdex.h
```
Comment (kcgen) [inline src/dos/Makefile.am:10]: If we're touching this Makefile, might as well clean it up into a sorted one-file-per-row list (similar to `include/Makefile.am`).

**Review by kcgen — CHANGES_REQUESTED** [review]

**Comment by kcgen** [comment]
Regarding: 

> Will probably need to add the contrib dir and other stuff too.

Feel free to propose adding `contrib` with another commit; it's a valuable addition for anyone who wants (or needs) to build from sources.

File: src/dos/Makefile.am (hunk 2 above)
Comment (drfiemost) [reply src/dos/Makefile.am]: Done!

**Comment by drfiemost** [comment]
> What does your workflow look like involving `make dist` and creating a source tarball?
> 

I'm using the standard autotools workflow: getting the source from the repo and running autogen.sh, then ./configure && make . After that I run make dist to create a tarball to keep the ready to use sources.
The make distcheck target is available too which checks that the tarball can be built without errors.

File: src/gui/Makefile.am (hunk 1 above)
Comment (drfiemost) [reply src/gui/Makefile.am]: Done! Also reindented and sorted items.

**Review by dreamer — APPROVED** [review]
@drfiemost  Changes are clean, `make dist` works correctly now, and you followed our commit message guidelines - perfect, thank you! :)

Personally, I don't use tarballs generated via `make dist` (I use tarballs generated by GitHub for us, e.g. https://github.com/dosbox-staging/dosbox-staging/archive/v0.75.2/whatevernameineedrightnow.tar.gz), but if we have `dist` target, then it should work correctly.

@kcgen Thanks for handling the review :)

**Comment by kcgen** [comment]
Thank you for the fix @drfiemost and final checks @dreamer.

**Pull request merged**